src/status_im2/contexts/chat/photo_selector/events.cljs

```diff
@@ -0,0 +1,144 @@
+(ns status-im2.contexts.chat.photo-selector.events
+  (:require [react-native.cameraroll :as cameraroll]
+            [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [utils.i18n :as i18n]
+            [react-native.permissions :as permissions]
+            [status-im2.config :as config]
+            [utils.re-frame :as rf]
+            [status-im.utils.image-processing :as image-processing]
+            [taoensso.timbre :as log]
+            [react-native.core :as rn]))
+
+(def maximum-image-size-px 2000)
+
+(defn- resize-photo
+  [uri callback]
+  (rn/image-get-size
+   uri
+   (fn [width height]
+     (let [resize? (> (max width height) maximum-image-size-px)]
+       (image-processing/resize
+        uri
+        (if resize? maximum-image-size-px width)
+        (if resize? maximum-image-size-px height)
+        60
+        (fn [^js resized-image]
+          (let [path (.-path resized-image)
+                path (if (string/starts-with? path "file") path (str "file://" path))]
+            (callback {:resized-uri path
+                       :width       width
+                       :height      height})))
+        #(log/error "could not resize image" %))))))
+
+(re-frame/reg-fx
+ :camera-roll-request-permissions-and-get-photos
+ (fn [[num end-cursor album]]
+   (permissions/request-permissions
+    {:permissions [:read-external-storage]
+     :on-allowed
+     (fn []
+       (cameraroll/get-photos
+        {:first      num
+         :assetType  (or end-cursor "Photos")
+         :groupTypes (if (= album (i18n/label :t/recent)) "All" "Albums")
+         :groupName  (when (not= album (i18n/label :t/recent)) album)
+         :include    (clj->js ["imageSize"])}
+        #(re-frame/dispatch [:on-camera-roll-get-photos (:edges %) (:page_info %) end-cursor])))})))
+
+(re-frame/reg-fx
+ :camera-roll-image-selected
+ (fn [[image chat-id]]
+   (resize-photo (:uri image) #(re-frame/dispatch [:photo-selector/image-selected chat-id image %]))))
+
+(defn get-albums
+  [callback]
+  (let [albums (atom {:smart-albums []
+                      :my-albums    []})]
+    ;; Get the "recent" album first
+    (cameraroll/get-photos
+     {:first 1 :groupTypes "All"}
+     (fn [res-recent]
+       (swap! albums assoc
+         :smart-albums
+         [{:title (i18n/label :t/recent)
+           :uri   (get-in (first (:edges res-recent)) [:node :image :uri])}])
+       ;; Get albums, then loop over albums and get each one's cover (first photo)
+       (cameraroll/get-albums
```

Comment:
I think there are some real misconceptions in all the code in this namespace in relation to how promises should be used, but it's not your PR's responsibility, so I won't comment on them.

Reply:
would be great to discuss, and i'll address in this PR no problem, i just don't know whats wrong

Reply:
yeah now i got it

Reply:
Sorry for the slow response @flexsurfer, it was a bit hard for me today to be responsive. It's hard to explain all the points about promises, and I didn't want to hold this PR from moving forward as well. This PR is already doing a bunch of good stuff, so it's totally fine by me if we ignore the whole refactoring of promises, and leave that for another moment.